content/modules/ROOT/pages/security-practices/ex1-signing-images.adoc

```diff
@@ -0,0 +1,572 @@
+    This can lead you to sign a different image than the intended one. Please use a
+    digest (example.com/ubuntu@sha256:abc123...) rather than tag
+    (example.com/ubuntu:latest) for the input to cosign. The ability to refer to
+    images by tag will be removed in a future release.
+
+
+        The sigstore service, hosted by sigstore a Series of LF Projects, LLC, is provided pursuant to the Hosted Project Tools Terms of Use, available at https://lfprojects.org/policies/hosted-project-tools-terms-of-use/.
+        Note that if your submission includes personal data associated with this signed artifact, it will be part of an immutable record.
+        This may include the email address associated with the account with which you authenticate your contractual Agreement.
+        This information will be used for signing this artifact and will be stored in public transparency logs and cannot be removed later, and is subject to the Immutable Record notice at https://lfprojects.org/policies/hosted-project-tools-immutable-records/.
+
+By typing 'y', you attest that (1) you are not submitting the personal data of any other person; and (2) you understand and agree to the statement and the Agreement terms at the URLs listed above.
+tlog entry created with index: 16
+Pushing signature to: quay-qkw52.apps.cluster-qkw52.dynamic.redhatworkshops.io/l3-students/l3-rhads-demoimage
+----
+
+[NOTE]
+====
+If you go to the `l3-rhads-demoimage` repository on {quay_url}/repository/l3-students/l3-rhads-demoimage?tab=tags[Quay,window="quay"], you can see that Quay recognises the `cosign` signature:
+
+image::security-practices/quay-signed-image.png[]
+
+That is a Quay feature - if or if not a customer's container registry will recognise signatures, depends.
+====
+
+
+
+
+
+=== More CI Integration
+
+Some CIs (such as Gitlab or Github) have a built-in authentication/access token that we can use. 
+
+If you have configured Trusted Artifact Signer with this OIDC system (if you followed the xref:setup-tas/setup-openshift.adoc[Setup Trusted Artifact Signer] exercise, you will remember that you can configure more than one OIDC system), `cosign` can directly use it. 
+
+In the examples below, you might wonder why we're not setting the `SIGSTORE_OIDC_ISSUER` environment variable (compare the output of the `help` statement in our terminal):
+.
+
+
+.*Click & Expand to see a GitHub Actions example*
+[%collapsible]
+====
+include::./ex1-signing-github.adoc[]
+====
+
+
+.*Click & Expand to see a GitLab CI example*
+[%collapsible]
+====
+include::./ex1-signing-gitlab.adoc[]
+====
+
+
+When using the CI's built-in identity token, you should NOT set `SIGSTORE_OIDC_ISSUER` - that would tell Cosign to do its own OIDC flow instead of using the CI token.
+The issuer is already embedded in the CI's OIDC token:
+
+* GitHub Actions: https://token.actions.githubusercontent.com
+* GitLab CI: https://gitlab.com (or your GitLab instance URL)
```

Comment:
When I click these explicit links they open in the same tab - so lose context. suggest making these asciidoc links like `https://gitlab.com[GitLab^]` (dont forget the caret!)